src/watertap_contrib/seto/property_models/basic_water_properties.py

```diff
@@ -0,0 +1,397 @@
+"""
+from idaes.core import (
+    EnergyBalanceType,
+    MaterialBalanceType,
+    MaterialFlowBasis,
+    PhysicalParameterBlock,
+    StateBlock,
+    StateBlockData,
+    declare_process_block_class,
+)
+from idaes.core.base.components import Solvent, Solute
+from idaes.core.base.phases import LiquidPhase
+from idaes.core.solvers.get_solver import get_solver
+from idaes.core.util.misc import add_object_reference
+from idaes.core.util.initialization import (
+    fix_state_vars,
+    revert_state_vars,
+    solve_indexed_blocks,
+)
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_unfixed_variables,
+)
+import idaes.logger as idaeslog
+import idaes.core.util.scaling as iscale
+from idaes.core.util.exceptions import InitializationError
+
+from pyomo.environ import (
+    Param,
+    PositiveReals,
+    units as pyunits,
+    Var,
+    Constraint,
+    Suffix,
+    value,
+    check_optimal_termination,
+)
+from pyomo.common.config import ConfigValue
+
+
+__author__ = "Kurban Sitterley"
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("BasicWaterParameterBlock")
+class BasicWaterParameterBlockData(PhysicalParameterBlock):
+    """
+    Property Parameter Block Class
+
+    Defines component lists, along with base units and constant
+    parameters.
+    """
+
+    CONFIG = PhysicalParameterBlock.CONFIG()
+
+    CONFIG.declare(
+        "solute_list",
+        ConfigValue(
+            domain=list,
+            description="List of solute species in the water source",
+        ),
+    )
+
+    def build(self):
+        """
+        Callable method for Block construction.
+        """
+        super().build()
+
+        self._state_block_class = BasicWaterStateBlock
+
+        self.Liq = LiquidPhase()
+
+        self.H2O = Solvent()
+
+        # Check definition of solute list
+        solute_list = self.config.solute_list
+
+        for j in solute_list:
+            self.add_component(str(j), Solute())
+
+        self.dens_mass = Param(
+            initialize=1000,
+            units=pyunits.kg / pyunits.m**3,
+            mutable=True,
+            doc="Mass density of flow",
+        )
+
+        self.visc_d = Param(
+            initialize=0.001,
+            units=pyunits.kg / pyunits.m / pyunits.s,
+            mutable=True,
+            doc="Dynamic viscosity of solution",
+        )
+        # ---------------------------------------------------------------------
+        # Set default scaling factors
+        self.set_default_scaling("temperature", 1e-3)
+        self.set_default_scaling("pressure", 1e-5)
+        self.set_default_scaling("dens_mass", 1e-3)
+        self.set_default_scaling("visc_d", 1e3)
+
+    @classmethod
+    def define_metadata(cls, obj):
+        obj.add_default_units(
+            {
+                "time": pyunits.s,
+                "length": pyunits.m,
+                "mass": pyunits.kg,
+                "amount": pyunits.mol,
```

Comment:
Just curious do we need all the basic units or we can remove "mol" if it's not used.